### PR TITLE
Update and extend docs for JavaScript, CSS, and HTML

### PR DESCRIPTION
**Pull request by devongovett**

### CODE REVIEW

File: src/languages/javascript.md [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+```javascript
+exports.multiply = function(a, b) {
+  return a * b;
+};
+```
+
+{% endsamplefile %}
+{% endsample %}
+
+In addition to `require` and `exports`, Parcel also supports `module.exports`, as well as the `__dirname` and `__filename` module globals, and `process.env` for access to environment variables. See the [Node emulation](/features/node-emulation/) docs for more details.
```
Comment (devongovett) [inline src/languages/javascript.md:78]: I'm debating whether the node emulation docs should actually be merged into this page since they are JS-specific

**Review by DeMoorJasper — APPROVED** [review]

File: src/languages/javascript.md [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+### JSX
+
+Parcel supports JSX out of the box. JSX is automatically enabled in a `.jsx` or `.tsx` file, or when one of the following libraries is defined as a dependency in your package.json:
+
+- `react`
+- `preact`
+- `nervjs`
+- `hyperapp`
+
+The correct JSX pragma is also automatically inferred based on the library you use. If none of the above libraries are installed, then the pragma defaults to `React.createElement`. Parcel also automatically detects the version of React or Preact that is installed, and enables the [modern JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) if supported.
```
Comment (mischnic) [inline src/languages/javascript.md]: > If none of the above libraries are installed, then the pragma defaults to `React.createElement`

But in that case, JSX won't be enabled automatically anyway?

File: src/languages/javascript.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+```
+
+{% endsamplefile %}
+{% endsample %}
+
+### Flow
+
+[Flow](https://flow.org) support is automatically enabled when `flow-bin` is listed as a dependency in your project's root package.json. You must also use a `@flow` directive in the files you wish to be compiled.
+
+Parcel currently uses Babel to strip flow types. If you have a custom Babel config, you will need to add the Flow plugin yourself. See Babel for more details.
```
Comment (mischnic) [inline src/languages/javascript.md]: > See Babel for more details.

Should this be a link?

File: src/languages/css.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+body {
+  background: url('images/background.png');
+}
+```
+
+Referenced files should be [relative](/features/dependency-resolution/#relative-specifiers) to the containing CSS file. You can also use [absolute](/features/dependency-resolution/#absolute-specifiers) and [tilde](/features/dependency-resolution/#tilde-specifiers) specifiers. The `data-url:` scheme can also be used to inline a file as a data URL. See [Bundle inlining](/features/bundle-inlining/) for more details.
+
+```css
+.logo {
+  background: url(data-url:./logo.png);
```
Comment (mischnic) [inline src/languages/css.md]:
```suggestion
  background: url('data-url:./logo.png');
```

Comment (devongovett) [reply src/languages/css.md]: pretty sure quotes aren't required, but I'll add them to match the other example

File: src/languages/javascript.md (hunk 2 above)
Comment (devongovett) [reply src/languages/javascript.md]: yeah I guess I was getting at when you used a `.jsx` or `.tsx` file but didn't have one of those libraries installed

Comment (devongovett) [reply src/languages/javascript.md]: but it's kinda confusing in this paragraph so I'll remove it

**Pull request merged**